### PR TITLE
Php diff 60

### PR DESCRIPTION
**Pull request by JBlond**

### CODE REVIEW

File: lib/jblond/Diff/Renderer/Text/UnifiedCli.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ public function render(): string
         if (!isset($this->options['cliColor'])) {
             return $this->output();
         }
-        if (isset($this->options['cliColor']) && $this->options['cliColor'] == 'simple') {
+        if (isset($this->options['cliColor']) && $this->options['cliColor'] === true) {
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/UnifiedCli.php:52]: I think `if ($this->options['cliColor']) {` is sufficient.

The variable should be set already by MainRendererAbstract and acting on any truthy value is ok.

File: lib/jblond/Diff/Renderer/Text/UnifiedCli.php [hunk 2]
```diff
@@ -63,7 +63,7 @@ public function render(): string
      */
     private function colorizeString($string, $color = ''): string
     {
-        if (isset($this->options['cliColor']) && $this->options['cliColor'] == 'simple') {
+        if (isset($this->options['cliColor']) && $this->options['cliColor'] === true) {
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/UnifiedCli.php:66]: I think `if ($this->options['cliColor']) {` is sufficient.

The variable should be set already by MainRendererAbstract and acting on any truthy value is ok.

**Review by DigiLive — COMMENTED** [review]
I've made some inline comments.

File: lib/jblond/Diff/Renderer/Text/UnifiedCli.php (hunk 2 above)
Comment (JBlond) [reply lib/jblond/Diff/Renderer/Text/UnifiedCli.php]: I changed that.

File: lib/jblond/Diff/Renderer/Text/UnifiedCli.php (hunk 1 above)
Comment (JBlond) [reply lib/jblond/Diff/Renderer/Text/UnifiedCli.php]: I changed that

File: lib/jblond/Diff/Renderer/Text/UnifiedCli.php [hunk 3]
```diff
@@ -49,10 +49,7 @@ public function render(): string
         if (!isset($this->options['cliColor'])) {
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/UnifiedCli.php:49]: `$this->options['cliColor']` should be set by MainRendererAbstract.
Therefor this line will always evaluate to true.

Please also check remaining code which uses this variable.

Comment (JBlond) [reply lib/jblond/Diff/Renderer/Text/UnifiedCli.php:49]: It is not always true. I check that via the example/cli.php script.

Comment (DigiLive) [reply lib/jblond/Diff/Renderer/Text/UnifiedCli.php:49]: That's why I'm saying 'should' while it's not. 😉
When extending `MainRendererAbstract`, the options for a sub renderer are supposed to be a merge of: 

- `MainRendererAbstract::$mainOptions` (Default renderer options)
- `SubRenderer::$subOptions` (Default child renderer options)
- The array parameter of `SubRenderer::__construct()` (custom options).

`SubRenderer::__construct()` should look something like:
```php
    public function __construct(array $options = [])
    {
        parent::__construct();
        $this->setOptions($this->subOptions); // Override/Add options MainRenderer with defaults of SubRenderer.
        $this->setOptions($options); // Override/Add options MainRenderer with customs.
    }
```
Method `setOptions`, inherited from `MainRendererAbstract`, will take care of the merges.

Currently `UnifiedCli::__construct()` is different:
```php
    public function __construct(array $options = [])
    {
        parent::__construct($options);
        $this->colors = new CliColors();
        $this->options = $options;
    }
````

You can look at `InlineCli.php` as an example on how to apply options.
Also it's described at https://github.com/JBlond/php-diff/wiki/b)-Parameters-and-Options#parameters-1
and https://github.com/JBlond/php-diff/wiki/b)-Parameters-and-Options#text-unified-cli-renderer

**Comment by DigiLive** [comment]
Please also check remaining code which uses `$this->options`.
Line 57-59 can be removed?
Line 71 always evaluates to true? Remove isset function?

**Comment by JBlond** [comment]
> Please also check remaining code which uses $this->options.
I did.

> Line 57-59 can be removed?
Yes

> Line 71 always evaluates to true? Remove isset function? 
Nope, in doubt the colorizeString function returns the string as is.

**Comment by DigiLive** [comment]
> Nope, in doubt the colorizeString function returns the string as is.

I'm not sure what 'In doubt' means.

Currently:
- $string is colorized when the option is set (Which what MainRenderAbstract will do), no matter the value is a truthy or falsy.
- $string is returned as is when the option isn't set (But it will be set by `MainRendererAbstract::__construct()` calling `MainRendererAbstract::setOptions()`).

Expected:
- $string is colorized when the option is set and the value is a truthy.
- $string is returned as is when the option isn't set or when the value is a falsy.

Having `if ($this->options['cliColor']) {...`
The way I see it, $this->options['cliColor'] is either set (by the MainRenderer) and its value is a truthy or falsy, or it isn't set which evaluates to a falsy also. (https://3v4l.org/vGsYB)

**Review by DigiLive — APPROVED** [review]
I think it's all ok now.
Maybe we should  user ask/warn vendeeglobe there's a PR merge comming up and ask for his feedback.

**Comment by JBlond** [comment]
@vendeeglobe please give feedback to this PR.

**Comment by vendeeglobe** [comment]
Looks good to me. I'll pull and test it after it's merged into master.

**Pull request merged**